Script.js

```diff
@@ -1,64 +1,136 @@
+  },
+];
+
+let currentQuestionIndex = 0;
+
 document.addEventListener("DOMContentLoaded", function () {
-  const quizForm = document.getElementById("quizForm");
-
-  quizForm.addEventListener("submit", function (event) {
-    event.preventDefault();
-
-    // Get the question text
-    const question = document.getElementById("question").value;
-
-    // Get all the options and their correctness
-    const options = Array.from(
-      document.querySelectorAll('input[name="option"]')
-    ).map((input, index) => ({
-      text: input.value,
-      isCorrect: document.querySelector(
-        `input[name="correct"][value="${index}"]`
-      ).checked,
-    }));
-
-    // Create the quiz question object
-    const quizQuestion = {
-      id: Date.now(), // simple unique ID based on timestamp
-      question: question,
-      options: options,
-      explanation: "",
-    };
-
-    // Output the quiz question object to console
-    console.log(quizQuestion);
-    alert("Quiz question created! Check console for output.");
-  });
-
-  // Add event listener to each radio button to update colors
-  document.querySelectorAll('input[name="correct"]').forEach((radio) => {
-    radio.addEventListener("change", updateColors);
-  });
-
-  // Function to update colors of options based on correctness
-  function updateColors() {
+  const quizContainer = document.getElementById("quizContainer");
+  const explanationDiv = document.getElementById("explanation");
+
+  function displayQuiz() {
+    quizContainer.innerHTML = ""; // Clear previous content
+    explanationDiv.textContent = ""; // Clear previous explanation
+
+    const questionElement = document.createElement("h2");
+    questionElement.textContent = quizQuestions[currentQuestionIndex].question;
+    quizContainer.appendChild(questionElement);
+
+    quizQuestions[currentQuestionIndex].options.forEach((option, index) => {
+      const optionContainer = document.createElement("div");
+      optionContainer.classList.add("option-container");
+
+      const radioInput = document.createElement("input");
+      radioInput.type = "radio";
+      radioInput.name = "quizOption";
+      radioInput.value = index;
+      radioInput.id = "option" + index;
+
+      const label = document.createElement("label");
+      label.htmlFor = "option" + index;
+      label.textContent = option.text;
+
+      optionContainer.appendChild(radioInput);
+      optionContainer.appendChild(label);
+      quizContainer.appendChild(optionContainer);
+    });
+  }
+
+  function randomizeOptions() {
+    quizQuestions[currentQuestionIndex].options.sort(() => Math.random() - 0.5);
+    displayQuiz();
+  }
+
+  function previousQuestion() {
+    if (currentQuestionIndex > 0) {
+      currentQuestionIndex--;
+      displayQuiz();
+    }
+  }
+
+  function nextQuestion() {
+    if (currentQuestionIndex < quizQuestions.length - 1) {
+      currentQuestionIndex++;
+      displayQuiz();
+    }
+  }
+
+  function submitQuiz() {
```

Comment:
Great how you have abstracted your logic into separate functions as well as creating the re-useable displayQuiz function